src/apps/spice-finance/ethereum/spice-finance.weth.token-fetcher.ts

```diff
@@ -0,0 +1,81 @@
+import { Inject } from '@nestjs/common';
+
+import { APP_TOOLKIT, IAppToolkit } from '~app-toolkit/app-toolkit.interface';
+import { PositionTemplate } from '~app-toolkit/decorators/position-template.decorator';
+import { Erc20, Erc721, Erc4626 } from '~contract/contracts';
+import { AppTokenTemplatePositionFetcher } from '~position/template/app-token.template.position-fetcher';
+import {
+  GetAddressesParams,
+  DefaultAppTokenDefinition,
+  GetUnderlyingTokensParams,
+  UnderlyingTokenDefinition,
+  GetDataPropsParams,
+  GetPricePerShareParams,
+  DefaultAppTokenDataProps,
+} from '~position/template/app-token.template.types';
+
+import { SpiceFinanceContractFactory, SpiceFinanceNftVault } from '../contracts';
+
+@PositionTemplate()
+export class EthereumSpiceFinanceWethTokenFetcher extends AppTokenTemplatePositionFetcher<SpiceFinanceNftVault> {
```

Comment:
I'm not quite sure I understand what is this suppose you want to show ? 
The contract you extend `AppTokenTemplatePositionFetcher<_HERE_>` has to be ERC20 compatible.

Reply:
Let me update it

Reply:
However, our vault token is not ERC20 but ERC721
May I use ERC721 instead of ERC20 then?

Reply:
@wpoulin

Reply:
What's the position/balances you want to show on Zapper ? The collection is already supported on Zapper https://zapper.xyz/nft/ethereum/0x6110d61dd1133b0f845f1025d6678cd22a11a2fe

Reply:
We are gonna show total WETH locked in the vault.

Reply:
You don't need to create an app token on top of the vault for weth to show up when you enter the wallet address on Zapper. Did I miss something ?

Reply:
We'd like to show as a protocol.

Reply:
ahh ok ok get it, it's mostly for showing up your TIV